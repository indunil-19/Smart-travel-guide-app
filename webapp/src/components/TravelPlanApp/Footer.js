import { Flex, HStack, VStack,Text ,Heading,Divider} from "@chakra-ui/layout"
import { Link } from "react-router-dom"

export const Footer=()=>{
    return(
        <>
        <Flex width="full" bg="black"  mt="165px" color="white" justifyContent="space-around" p={10}  >
             <VStack>
                <HStack spacing={10}>
                <VStack alignItems="start" spacing={8}>

                     
                     
                     <Heading as="h3" size="lg">
                        About Us
                    </Heading>
                 
                    

                    <Heading as="h3" size="lg">
                        Contact Us
                    </Heading>
                  
                </VStack>

                <Divider orientation="vertical"   variant="solid" p={3}  />

                <VStack>
                     
                     <>

                     <iframe src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId" width="340" height="500" style={{border:"none",overflow:"hidden"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>                     </>
                </VStack>
                </HStack>
                       <Divider  orientation="horizontal"   variant="solid" p={3}  />
                </VStack>
               
        </Flex>
       


        </>
    )
}