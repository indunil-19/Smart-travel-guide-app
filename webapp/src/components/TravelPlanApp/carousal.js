import React, { useEffect,useState } from "react";
// import  "react-responsive-carousel/lib/styles/carousel.min.css"; 
import { Carousel } from 'react-responsive-carousel';




 const Carousal=({photos=[]})=>{
    const [references, setReferences]=useState([])
    useEffect(()=>{
            setReferences(photos)
    }, [photos])
    return(
        <Carousel>
                
                {
                    references.map((reference)=>{
                        return(
                            <div >
                            <img data-testid="image"  src={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=800&photo_reference=${reference.photo_reference}&key=AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA`} 
                             />
                            
                        </div>
                        )
                    })
                }
                
            </Carousel>
    )
}
export default Carousal