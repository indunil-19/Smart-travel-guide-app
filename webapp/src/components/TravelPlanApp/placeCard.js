import React , {useEffect,useState}from "react"
import { useHistory } from "react-router"
import { Box, Flex ,HStack, VStack} from "@chakra-ui/layout"
import { Image,Button,Badge, Avatar , Heading} from "@chakra-ui/react"
import { StarIcon } from "@chakra-ui/icons"
import StarRatings from "react-star-ratings"

export const PlaceCard=({photo="",index,name,address, types=[], rating, place_id})=>{
    const [link,setLink]=useState("")
    const history=useHistory()
    useEffect(() => {
        
        
        setLink(photo)
    }, [photo])
    return(
        <Flex flexDirection="row" margin={5} borderWidth="1px" borderRadius="lg" p={5} width="100%">
                           <Image src={link}  width="400px" height="300px" objectFit="cover" borderRadius="lg"  />
                           <VStack spacing={3.5} alignItems="start" p={5}>
                                 <Box bg="green.200" p={3} borderRadius="full" border="green" borderWidth="1px" ><Heading as="h4" size="md">{index.toString()}</Heading></Box>
                                 <Box lineHeight="tall" fontWeight="semibold" as="h4"> {name}</Box>
                                 <Box color="gray.600" fontSize="sm">{address}</Box>
                                <HStack>
                                        <>    {types.map((item)=>{
                                                return(
                                                    <Badge borderRadius="full" px="2" colorScheme="teal">{item}</Badge>
                                                )
                                            })}
                                        </> 
                                         
                                </HStack>

                                <HStack>
                                        {/* <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} /> */}


                                        <StarRatings
                                        rating={rating}
                                        starRatedColor="teal"
                                        numberOfStars={5}
                                        name='rating'
                                        starDimension="20px"
                                        starSpacing="3px"
                                        />
                                        <Box>rating {rating}</Box>
                                        
                                </HStack>
                               
                                        <Button colorScheme="teal" variant="outline" onClick={()=>{
                                            history.push('/travelPlan/viewpoi/'+place_id)
                                        }}>
                                            See more...
                                        </Button>

                           </VStack>

        </Flex>
    )
}