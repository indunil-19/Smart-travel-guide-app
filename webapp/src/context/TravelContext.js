import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 