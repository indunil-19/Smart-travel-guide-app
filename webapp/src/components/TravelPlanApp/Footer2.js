import { Flex ,Text } from "@chakra-ui/layout"
import React from "react"
import Footer from "./Footer"

 const Footer2=()=>{
    return(
        <>
        <Flex width="full" bg="black" color="white"  p={5} flexDirection="column" alignItems="center" >

              <Text fontSize="xs">@university of moratuwa 2021/09/27</Text>


        </Flex>
        </>
    )
}
export default Footer2