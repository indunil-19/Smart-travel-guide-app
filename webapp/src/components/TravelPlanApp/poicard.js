import { Divider, Flex, HStack, Text,  } from "@chakra-ui/layout"
import React from "react"
import { Heading } from "@chakra-ui/layout";
import { IoLocation } from "react-icons/io5";
import { Tag } from "@chakra-ui/tag";
import { Carousal } from "../../components/TravelPlanApp/carousal"
import { Review } from "../../components/TravelPlanApp/review"


 const PoiCard=({data={}})=>{
    return(
        <Flex  flexDirection="column" alignItems="center" width="70%" marginX="auto" boxShadow="dark-lg" padding={4} marginTop={2}>            
            

            <HStack><IoLocation/>
            <Heading padding={5}>{data.name}</Heading>
            </HStack>

            <Carousal photos={data.photos} />

            <HStack spacing={6}  p={2}> 
                {(data.types)  && data.types.map((type)=>{
                    return(
                        <Tag  variant="solid" colorScheme="teal">
                            {type}
                        </Tag>
                    )
                })}
            </HStack>

            <Text fontSize="md">{data.formatted_address}</Text>
            
         




            <Divider  padding="5" />
            
            <Text fontSize="3xl">User reviews</Text>
            
            <>
            {(data.reviews) && data.reviews.map((review)=>{
                return(
                    <Review  author_name={review.author_name} profile_photo_url={review.profile_photo_url} relative_time_description={review.relative_time_description} text={review.text} rating={review.rating} author_url={review.author_url}/>
                )
            })}
            
            
            </>
                       
        </Flex>
    )
}
export default PoiCard