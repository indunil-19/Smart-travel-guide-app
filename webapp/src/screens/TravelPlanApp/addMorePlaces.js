import { useContext, useState, useEffect } from "react"
import { findPois, addPoiToPlan } from "../../services/EditPlanServices"
import { useParams } from 'react-router'
import { TravelContext } from "../../context/TravelContext"
import { PlaceCard } from "../../components/TravelPlanApp/placeCard"
import { Flex, HStack ,VStack} from "@chakra-ui/layout"
import { GrAddCircle } from "react-icons/gr";
import { Button } from "@chakra-ui/button"
import { useHistory } from "react-router"
import { Tabs, TabList, TabPanels, Tab, TabPanel,Alert} from "@chakra-ui/react"
import { AlertIcon } from "@chakra-ui/alert"
import Autocomplete from "react-google-autocomplete";

export const AddMorePlaces=()=>{
    const history=useHistory()
    const {day}=useParams()
    const [pois,setPois]=useState([])
    const [route,setRoute]=useState([])
    const {state, dispatch}=useContext(TravelContext)
    const [place1,setPlace1]=useState({})


    const [poisCustom,setPoisCustom]=useState([])
    const [routeCustom,setRouteCustom]=useState([])


    useEffect(() => {
        findPois(day, state.travelPlan, state.allpois).then((res)=>{
            console.log(res)
            setPois(res[0])
            setRoute(res[1])
        })
    }, [state])


    const addPoi=(day,poi, route)=>{
            addPoiToPlan(day,poi, route,state.travelPlan).then((res)=>{
                // console.log(res)
                dispatch({type:"set_travelPlan" , payload:{travelPlan:res}})
            })
    }

    return(
        <>
        <Tabs isFitted variant="line"  >
        <TabList  position="fixed" width="full" zIndex="8" bg="white" p={2} >
            <Tab >Place sugests for you</Tab>
            <Tab>Find by your self</Tab>
        </TabList>


        <TabPanels >


        <TabPanel>
        <Flex flexDirection="column" alignItems="center">
        {
            pois.length==0 ? 
            <>
                <Alert status="warning" mt="100px">
                    <AlertIcon />
                    Seems there is no pois for suggests. 
                </Alert>
            
            </> :<></>
        }
        
        { 
         pois.map((Item, index)=>{
            return(
                <>
                <HStack m={3}>

                <Button colorScheme="black" variant="outline" onClick={()=>{
                                addPoi(day,Item, route[index]);
                                history.push("/travelPlan/editPlan")

                 }}>
                     <GrAddCircle />           
                </Button>



                <PlaceCard  index={index+1} name={Item.name} address={Item.vicinity?Item.vicinity : Item.formatted_address  } photo={Item.photos ? Item.photos[0].photo_reference : "" } rating={Item.rating} place_id={Item.place_id}/>
                </HStack>
                </>
            )
        })

    }



        
        
        </Flex>
        </TabPanel>


        <TabPanel>


        <Flex flexDirection="column" alignItems="center" p={10}>
                <Autocomplete style={{width:"40%" ,height:"30px", padding:"5px", margin:"15px",background:"grey", borderRadius:"5px", color:"white"}}
                apiKey={"AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA"}

                onPlaceSelected={(place) => {
                if(place.geometry.location){
                    setPlace1(place)
                    findPois(day, state.travelPlan, [place]).then((res)=>{
                        // console.log(res)
                        setPoisCustom(res[0])
                        setRouteCustom(res[1])
                    })
                }
                }}

                options={{
                    types: ['geocode' , 'establishment'],
                    componentRestrictions: { country: "LK" },
                    fields:["ALL"]
                }}
                >
                </Autocomplete>


        </Flex>






        </TabPanel>



        </TabPanels>
        </Tabs>
        </>
    )
}