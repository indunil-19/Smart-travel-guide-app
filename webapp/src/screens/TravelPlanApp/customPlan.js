import { Flex,VStack,HStack,Text,} from '@chakra-ui/layout';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"
import {useState,useEffect,useContext,useRef} from "react";
import Autocomplete from "react-google-autocomplete";
import { useHistory } from "react-router"
import { TravelContext } from "../../context/TravelContext"
import { Button, Heading,Image,AlertDialog,AlertDialogBody,AlertDialogFooter,AlertDialogHeader,AlertDialogContent,AlertDialogOverlay} from "@chakra-ui/react"
import { MdDeleteForever } from "react-icons/md";
import { calculateAndDisplayRoute } from '../../services/TravelPlanService';

export const CustomPlanInner=()=>{
  const history=useHistory()
  const {state, dispatch}=useContext(TravelContext)

  const [p,setP]=useState([])
 
 
  const [isOpen, setIsOpen] = useState(false)
  const onClose = () => setIsOpen(false)
  const cancelRef = useRef()
  

    
    
    return(
        <>
      <AlertDialog
                isOpen={isOpen}
                leastDestructiveRef={cancelRef}
                onClose={onClose}
              >
                <AlertDialogOverlay>
                  <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                      Places error
                    </AlertDialogHeader>
        
                    <AlertDialogBody>
                      you can't add more than 24 places to your plan
                    </AlertDialogBody>
        
                    <AlertDialogFooter>
                      <Button ref={cancelRef} onClick={onClose}>
                        Cancel
                      </Button>
                    </AlertDialogFooter>
                  </AlertDialogContent>
                </AlertDialogOverlay>
              </AlertDialog>



       <Flex width="100%" alignItems="center" flexDirection="column" justifyContent="center">
                  
        <Autocomplete style={{width:"40%" ,height:"30px", padding:"5px", margin:"15px",background:"grey", borderRadius:"5px", color:"white"}}
        apiKey={"AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA"}

        onPlaceSelected={(place) => {
          if(place.length>24){
              setIsOpen(true)
          }



          if(place.geometry.location){
            // console.log(place.photos[0].getUrl())
            setP(p=>[...p,place])
          }
          


        }}

        options={{
            types: ['geocode' , 'establishment'],
            componentRestrictions: { country: "LK" },
            fields:["ALL"]
        }}
        >
        </Autocomplete>

           
      </Flex>
      <Flex justifyContent="center" flexDirection="column" alignItems="center">
     
        {p.map((i,index)=>{
          
          return(
            <>
            
            <HStack p={3}>
            <Heading as="h5" size="md">
                {index+1}
                {".  "}
                {i.name}
            </Heading>

            <Button onClick={()=>{
                  console.log(index)
                  setP(p.filter((item,index1)=>{
                       if(index1 !==index)  return item
                  }))
                 }}>
                 <MdDeleteForever />
            </Button>
            </HStack>
            </>
          )
         
        })}


            <GoogleMap
                    defaultZoom={7}
                    defaultCenter={{ lat: 7.291418, lng:80.636696}}
                  >
            
                    {p && p.map((item=[],index)=>{
                        return(
                            <>
                               
                                        <>
                                        <Marker position={item.geometry.location} label={`${index+1}`} />
                                        
                                        </>
                            </>
                        )
                    })

                    }     
    
            </GoogleMap>




        {
          p.length ?

          <Button colorScheme="teal" variant="outline" onClick={
           ()=>{
            calculateAndDisplayRoute(p).then(r=>{
             
              dispatch({type:"set_travelPlan" , payload:{travelPlan:r[0]}})
              dispatch({type:"set_pois" , payload:{allpois:r[1]}})
              history.push("/travelPlan/travelPlan")

            })
           }
          }>
          submit
        </Button> :
        <></>

        }
        </Flex>




        </>
    )
}
const MapWrapped = withScriptjs(withGoogleMap(CustomPlanInner));

export const CustomPlan=()=>{
  return(
    <Flex width="80%" height="180vh" flexDirection="column" mx="auto" my="2" boxShadow="dark-lg">
        <MapWrapped
          googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA`}
          loadingElement={<div style={{ height: `100%` }} />}
          containerElement={<div style={{ height: `100%` }} />}
          mapElement={<div style={{ height: `40%` }} />}
        />
        </Flex>    
  )
}


