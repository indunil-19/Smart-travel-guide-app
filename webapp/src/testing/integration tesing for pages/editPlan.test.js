import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import { createMemoryHistory } from 'history'
import { EditPlan,Card } from "../../screens/TravelPlanApp/editPlan";


describe('edit plan page', () => {
    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><EditPlan/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><EditPlan  /></TravelProvider></MemoryRouter>,div)
    })

    

})
