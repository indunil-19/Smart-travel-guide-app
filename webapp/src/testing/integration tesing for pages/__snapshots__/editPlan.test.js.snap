// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit plan page should match snapshot 1`] = `
<div
  className="css-dvxtzn"
>
  <div
    className="chakra-stack css-84zodg"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z"
      />
    </svg>
    <span
      className="chakra-badge css-ifsvay"
    >
      9.00 a.m
    </span>
    <p
      className="chakra-text css-1tapsqs"
    >
      Start from colombo
    </p>
  </div>
  <button
    className="chakra-button css-f37c8w"
    onClick={[Function]}
    type="button"
  >
    Add A Day
  </button>
  <div
    className="chakra-stack css-84zodg"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z"
      />
    </svg>
    <span
      className="chakra-badge css-ifsvay"
    >
      6.00 p.m
    </span>
    <p
      className="chakra-text css-1tapsqs"
    >
      End - 
      
    </p>
  </div>
  <div
    className="chakra-stack css-1obfgbw"
  >
    <button
      className="chakra-button css-xdqcq1"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
        />
      </svg>
    </button>
  </div>
</div>
`;
